### ISSUE TITLE
Add Future::flatten

### ISSUE THREAD

**Issue by yoshuawuyts** [async-rs/async-std#404]
Similar to future-rs's [`Future::flatten`](https://docs.rs/futures-preview/0.3.0-alpha.19/futures/future/trait.FutureExt.html#method.flatten) method, this can sometimes come in useful.

However the only difference with it should be that we take `IntoFuture` as the trait bound, and call it under the hood. Because `IntoFuture` is implemented for `Future` this should work as expected.

Thanks!

## Examples
```rust
let nested_future = async { async { 1 } };
let future = nested_future.flatten();
assert_eq!(future.await, 1);
```

**Comment by portgasd666** [comment]
I can take a stab at this

**Comment by skade** [comment]
@portgasd666 go ahead!

**Comment by k-nasa** [comment]
This seems to have been resolved! by #443
